src/external/abinit.jl

```diff
@@ -0,0 +1,128 @@
+using PyCall
+import JSON
+
+"""
+Run an SCF in ABINIT starting from the input file `infile`
+represented as a `abipy.abilab.AbinitInput` python object.
+Write the result to the `output` directory in ETSF Nanoquanta format
+and return the result as an `EtsfFolder` object.
+"""
+function run_abinit_scf(infile::PyObject, outdir)
+    abilab = pyimport("abipy.abilab")
+    flowtk = pyimport("abipy.flowtk")
+
+    # Check ABINIT can be found
+    abierrors = abilab.abicheck()
+    abierrors != "" && error("abilab.abicheck() returned errors:\n$abierrors")
+
+    # Create rundir
+    rundir = joinpath(outdir, "abinit")
+    if isdir(rundir)
+        rm(rundir, recursive=true)
+    end
+    mkdir(rundir)
+
+    # Adjust common infile settings:
+    infile.set_vars(
+        paral_kgb=0,    # Parallelisation over k-Points and Bands
+        iomode=3,       # Use NetCDF output
+        istwfk="*1",    # Needed for extracting the wave function later
+    )
+
+    # Dump pspmap
+    open(joinpath(outdir, "pspmap.json"), "w") do fp
+        JSON.print(fp, infile.pspmap)
+    end
+    infile.extra = nothing  # Remove extra field (causes problems below)
+
+    # Create flow and run it
+    flow = flowtk.Flow(rundir, flowtk.TaskManager.from_user_config())
+    work = flowtk.Work()
+    scf_task = work.register_scf_task(infile)
+    flow.register_work(work)
+    flow.allocate()
+    flow.make_scheduler().start()
+
+    if !flow.all_ok
+        @warn "Flow not all_ok ... check input and output files"
+    end
+
+    for file in collect(scf_task.out_files())
+        if endswith(file, ".nc")
+            cp(file, joinpath(outdir, basename(file)), force=true)
+        end
+    end
+
+    EtsfFolder(outdir)
+end
+
+
+"""
+Run an SCF in ABINIT starting from a DFTK `Model` and some extra parameters.
+Write the result to the `output` directory in ETSF Nanoquanta format
+and return the `EtsfFolder` object.
+"""
+function run_abinit_scf(model::Model, outdir;
```

Comment:
Should we maybe make it so that PWbasis contains all the info needed for that? Eg add some psp metadata. Maybe also the nbands should be part of it...

Reply:
Point is it might be good to have a DFTK object that maps cleanly to "all that is needed to run a SCF in a PW code". Whether that's in PWBasis or somewhere else I'm not sure

Reply:
With the last point I agree, with the first I'm not too sure. For me this interface to ABINIT is just convenience and I do not think this should guide our design. The psp thing could be solved for example by being able to write or read ABINIT psp files, but I just could not be bothered to implement that for now ...

Reply:
Having metadata (eg what the psp file was) can't hurt, I'm keeping that in mind for the refactor. OK, let's keep it as is. PWbasis = discretization only. Solving the discretized equations is another matter.

Reply:
> Having metadata (eg what the psp file was) can't hurt

But that is already there: Inside the atoms. It's just in this case I write stuff to disk in ABINIT format, where the DFTK pseudos are lost and then I was lazy.

Reply:
Ah I see; never mind then!